File: tool_code/read_file.js
async function read_file(input) {
    const fs = require('fs').promises;
    const path = require('path');

    const pathSanitizer = (filePath) => {
        return path.normalize(filePath.split('\\').join('/')).replace(/\/+/g, '/');
    };
    if (!input || typeof input.file_path !== 'string') {
        console.error('❌ Invalid input values.');
        return false;
    }

    let { file_path } = input;
    file_path = pathSanitizer(file_path);

    try {
        const content = await fs.readFile(file_path, 'utf8');
        console.log('✅ File read successfully: ' + file_path);
        console.log(content);
        return content;
    } catch (err) {
        console.error('❌ Error reading file: ' + file_path, err);
        return false;
    }
}

File: tool_code/read_file.py
import os
import re
def path_sanitizer(file_path):
    normalized = os.path.normpath(file_path.replace('\\', '/'))
    return re.sub(r'/+', '/', normalized)
def read_file(input_data):
    if not input_data or not isinstance(input_data.get('file_path'), str):
        print('❌ Invalid input values.')
        return ''
    file_path = path_sanitizer(input_data['file_path'])
    try:
        with open(file_path, mode='r', encoding='utf8') as f:
            content = f.read()
            print(f'✅ File read successfully: {file_path}')
            print(content)
            return content
    except Exception as err:
        print(f'❌ Error reading file: {file_path}', err)
        return ''